angular-workspace/projects/ni/nimble-angular/table-column/anchor/nimble-table-column-anchor-navigation-guard.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import type { TableColumnAnchorCellView } from '@ni/nimble-components/dist/esm/table-column/anchor/cell-view';
+
+export type ClickDelegate = (rowRecordId: string | undefined) => boolean;
+
+/**
+ * Directive to allow client to intercept anchor clicks and do router navigation instead of
+ * letting the default action handle the navigation.
+ */
+@Directive({
+    selector: 'nimble-table-column-anchor[clickDelegate]'
```

Comment:
the directive should match the directive name and clean-up language around "click delegate". From the angular context they don't care about click / event delegates. They just care about being able to service routing themselves or letting the link do it's own thing.

```
    selector: 'nimble-table-column-anchor[navigationGuard]'
```